### ISSUE TITLE
More polished unindent for curly braces

### ISSUE THREAD

**Issue by olafurpg** [olafurpg/scalafmt#394]
```diff
      f(
          (future, message) ⇒
            (evaluating {
                  Await.result(future.mapTo[java.lang.Thread],
                               timeout.duration)
                } should produce[java.lang.Exception]).getMessage should ===(
                message))

    assert(
        backoffs.force.toSeq ==
          (0 until 10 map { i =>
                (1 << i).seconds
              }))
      f(
          (future, message) ⇒
            (evaluating {
                  Await.result(future.mapTo[java.lang.Thread],
                               timeout.duration)
                } should produce[java.lang.Exception]).getMessage should ===(
                message))
    assert((p1 match {
          case Param.Configured(x) => x()
          case x => throw new MatchError(x)
        }) == failureAccrualPolicy)
      setBox(Full((array.map {
                case JsonObjectId(objectId) => objectId
                case JsonRegex(regex) => regex
                case JsonUUID(uuid) => uuid
                case JsonDateTime(dt)
                    if (mf.toString == "org.joda.time.DateTime") =>
                  dt
                case JsonDate(date) => date
                case other => other.values
              }).asInstanceOf[MyType]))
    def bindingInfoWithFields(style: BindingStyle) =
      logger.trace("Looking for fields with style %s".format(style), (for {
            field <- fields;
            bindingInfo <- field.binding if bindingInfo.bindingStyle == style
          } yield (bindingInfo, field)).toList)
    Html((filter match {
          case GameFilter.All =>
            info.user.count.game + " " + trans.gamesPlayed()
          case GameFilter.Me =>
            ctx.me ?? (me => trans.nbGamesWithYou.str(info.nbWithMe))
          case GameFilter.Rated => info.nbRated + " " + trans.rated()
          case GameFilter.Win => trans.nbWins(info.user.count.win)
          case GameFilter.Loss => trans.nbLosses(info.user.count.loss)
          case GameFilter.Draw => trans.nbDraws(info.user.count.draw)
          case GameFilter.Playing => info.nbPlaying + " playing"
          case GameFilter.Bookmark => trans.nbBookmarks(info.nbBookmark)
          case GameFilter.Imported => trans.nbImportedGames(info.nbImported)
          case GameFilter.Search =>
            Html(trans.advancedSearch.str().replaceFirst(" ", "\n"))
        }).toString)

                                  (Authorities
                                        .ifPresent(sreq.owners) \/> badRequest(
                                          "You must provide an owner account for the task results!"))

             parse0(tail, (head match {
-              case "[*]" => CPathArray
-              case IndexPattern(index) => CPathIndex(index.toInt)
+                  case "[*]" => CPathArray
+                  case IndexPattern(index) => CPathIndex(index.toInt)

-              case name => CPathField(name)
-            }) :: acc)
+                  case name => CPathField(name)
+                }) :: acc)

                           jv.validated[AddScheduledQueryRequest]
                             .disjunction leftMap { err =>
                             badRequest(
                                 "Request body %s is not a valid scheduling query request: %s"
                                   .format(jv.renderCompact, err.message))
                           }
-      val t = System.nanoTime() - t0
-      System.err.println("%s took %.2fms" format (s, t / m))
-      StreamT.empty[M, A]
-    }).point[M])
+          val t = System.nanoTime() - t0
+          System.err.println("%s took %.2fms" format (s, t / m))
+          StreamT.empty[M, A]
+        }).point[M])
     stream ++ end

     (ref.copy(selector = (prefix \ ref.selector)), (ref.ctype match {
-      case CString => ArrayStrColumn.empty(sliceSize)
-      case CBoolean => ArrayBoolColumn.empty()
-      case CLong => ArrayLongColumn.empty(sliceSize)
-      case CDouble => ArrayDoubleColumn.empty(sliceSize)
-      case CNum => ArrayNumColumn.empty(sliceSize)
-      case CDate => ArrayDateColumn.empty(sliceSize)
-      case CPeriod => ArrayPeriodColumn.empty(sliceSize)
-      case CNull => MutableNullColumn.empty()
-      case CEmptyObject => MutableEmptyObjectColumn.empty()
-      case CEmptyArray => MutableEmptyArrayColumn.empty()
-      case CArrayType(elemType) =>
-        ArrayHomogeneousArrayColumn.empty(sliceSize)(elemType)
-      case CUndefined => sys.error("CUndefined cannot be serialized")
-    }))
+          case CString => ArrayStrColumn.empty(sliceSize)
+          case CBoolean => ArrayBoolColumn.empty()
+          case CLong => ArrayLongColumn.empty(sliceSize)
+          case CDouble => ArrayDoubleColumn.empty(sliceSize)
+          case CNum => ArrayNumColumn.empty(sliceSize)
+          case CDate => ArrayDateColumn.empty(sliceSize)
+          case CPeriod => ArrayPeriodColumn.empty(sliceSize)
+          case CNull => MutableNullColumn.empty()
+          case CEmptyObject => MutableEmptyObjectColumn.empty()
+          case CEmptyArray => MutableEmptyArrayColumn.empty()
+          case CArrayType(elemType) =>
+            ArrayHomogeneousArrayColumn.empty(sliceSize)(elemType)
+          case CUndefined => sys.error("CUndefined cannot be serialized")
+        }))


   println((for (i <- 0 to 10) yield {
-    val in = ArrayStack.tabulate(i)(_.toString)
-    (in, (in filter (_ => true)) == in)
-  }).mkString("\n"))
+        val in = ArrayStack.tabulate(i)(_.toString)
+        (in, (in filter (_ => true)) == in)
+      }).mkString("\n"))


       assertEquals(1, ((NoContext: Any) match {
-        case that: AnyRef if this eq that => 0
-        case NoContext => 1
-        case _ => 2
-      }))
+            case that: AnyRef if this eq that => 0
+            case NoContext => 1
+            case _ => 2
+          }))

// OK
        b sort {
          BSONDocument(
              (for (sorter ← sorters)
                yield
                  sorter._1 -> BSONInteger(sorter._2 match {
                    case api.SortOrder.Ascending => 1
                    case api.SortOrder.Descending => -1
                  })).toStream)
        }
    copy(user1 = user1.copy(
             score =
               user1.score + (userId match {
                 case None => wins * 5
                 case Some(u) if user1.id == u => wins * 10
                 case _ => 0
               })),
         user2 = user2.copy(
             score =
               user2.score + (userId match {
                 case None => wins * 5
                 case Some(u) if user2.id == u => wins * 10
                 case _ => 0
               })))

           (new FastPath(_) {
-        val (a, b, c, d) = (next[Int], next[Int], next[Int], next[Int])
-        override def read(r: Reader) =
-          new A(a.read(r), b.read(r), c.read(r), d.read(r))
-      }))
+            val (a, b, c, d) = (next[Int], next[Int], next[Int], next[Int])
+            override def read(r: Reader) =
+              new A(a.read(r), b.read(r), c.read(r), d.read(r))
+          }))


         (e: Either[Int, Int], or: Int) =>
           e.left.getOrElse(or) ==
             (e match {
-          case Left(a) => a
-          case Right(_) => or
-        }))
+              case Left(a) => a
+              case Right(_) => or
+            }))

```

**Comment by olafurpg** [comment]
The indentation rules have changed for code like this

```scala
  assertEquals(1, ((NoContext: Any) match {
                     case NoContext => 1
                     case _ => 2
                   }))
  RedisCodec.toUnifiedFormat(
      Sequence(CommandBytes.BITCOUNT, key) ++
        (lst ++ (start match {
                   case None => Seq.empty
                 })))
```

I would like to achieve a layout like below but I haven't been successful yet.

```scala
  assertEquals(1, ((NoContext: Any) match {
    case NoContext => 1
    case _ => 2
  }))
  RedisCodec.toUnifiedFormat(
    Sequence(CommandBytes.BITCOUNT, key) ++
      (lst ++ (start match {
        case None => Seq.empty
      }))
```

**Comment by olafurpg** [comment]
Closed by 9d01993b4bdee0159664806dc59384a65568f8c3
